woocommerce/woocommerce-ios
# Product List: add Products tab under a feature flag

Fixes #1261 

Update release notes:


## Changes
- Added `productList` to `FeatureFlag` enabled only for debug builds
- Refactored `WooTab` enum so that its tab index is calculated instead of fixed raw value
  - Updated initializer to `init(visibleIndex: Int, isProductListFeatureOn: Bool)`
    - Also updated existing `WooTab` `rawValue` usage
  - Added `visibleIndex(isProductListFeatureOn: Bool) -> Int` for the index of a tab
- In `MainTabBarController`, configured Products tab if feature flag is enabled in `viewDidLoad` (all the other tab view controllers are set up in `Main.storyboard`)

## Testing

Case 1: feature flag is on
- Check out PR branch and build the app
- When logged in, there should be 4 tabs like in screenshots

Case 2: feature flag is off
- In `DefaultFeatureFlagService`, return false for `productList` case
- Build the app
- When logged in, there should be 3 tabs as before

## Example screenshots

\ | unselected | selected
-- | -- | --
portrait | ![Simulator Screen Shot - iPhone 5s - 2019-09-05 at 15 10 14](https://user-images.githubusercontent.com/1945542/64319827-d9387500-cfef-11e9-8bac-cbaf1c7416ba.png) | ![Simulator Screen Shot - iPhone 5s - 2019-09-05 at 15 10 16](https://user-images.githubusercontent.com/1945542/64319828-d9387500-cfef-11e9-8bff-4a3b303f9706.png)
landscape | ![Simulator Screen Shot - iPhone 5s - 2019-09-05 at 15 10 25](https://user-images.githubusercontent.com/1945542/64319860-e6556400-cfef-11e9-9eac-e247a90acb9d.png) | ![Simulator Screen Shot - iPhone 5s - 2019-09-05 at 15 10 27](https://user-images.githubusercontent.com/1945542/64319861-e6556400-cfef-11e9-9da6-8dcbf46a12ff.png)
